src/service-worker.js

```diff
@@ -0,0 +1,59 @@
+import { clientsClaim } from 'workbox-core';
+import { ExpirationPlugin } from 'workbox-expiration';
+import { precacheAndRoute, createHandlerBoundToURL } from 'workbox-precaching';
+import { registerRoute } from 'workbox-routing';
+import { StaleWhileRevalidate } from 'workbox-strategies';
+
+clientsClaim();
+
+precacheAndRoute(self.__WB_MANIFEST);
+
+const fileExtensionRegexp = new RegExp('/[^/?]+\\.[^/]+$');
+registerRoute(
+    // Return false to exempt requests from being fulfilled by index.html.
+    ({ request, url }) => {
+        // If this isn't a navigation, skip.
+        if (request.mode !== 'navigate') {
+            return false;
+        } // If this is a URL that starts with /_, skip.
+
+        if (url.pathname.startsWith('/_')) {
+            return false;
+        } // If this looks like a URL for a resource, because it contains // a file extension, skip.
+
+        if (url.pathname.match(fileExtensionRegexp)) {
+            return false;
+        } // Return true to signal that we want to use the handler.
+
+        return true;
+    },
+    createHandlerBoundToURL(process.env.PUBLIC_URL + '/index.html')
```

Comment:
Not sure if this will work at all by fetching this env var as it is not defined while building the application. Just provided on runtime.

Reply:
In a React application, the process.env.PUBLIC_URL contains the path to the `Public` folder where we keep our assets. This will work, since React actually injects this env variable and makes it available to be used when needed. 
In this particular line of code, we are telling the service-worker that the `index.html` in our build directory's `Public` folder is the entry into our application and that is the file we want to cache and render.

Please refer to this document for more details: 
https://create-react-app.dev/docs/using-the-public-folder/#adding-assets-outside-of-the-module-system

This file:


![Screenshot 2022-10-02 at 23 50 19](https://user-images.githubusercontent.com/19223383/193469691-227f684a-038a-4aa4-8bc5-e04af3ba5da1.png)

Reply:
ah, yes of course. I was mistaken of this ENV variable with the HOST variable set on runtime.